### ISSUE TITLE
Break out PR #35 into multiple smaller PR's

### ISSUE THREAD

**Issue by cdancy** [cdancy/jenkins-rest#48]
User declined the PR but I think we should leverage that work and create more targeted PR's to tackle each endpoint.

**Comment by martinda** [comment]
I agree with breaking PR #35 into multiple smaller PRs. It will be easier to review.
